core/primitives/block_header.hpp

```diff
@@ -48,8 +49,8 @@ namespace kagome::primitives {
   template <class Stream,
             typename = std::enable_if_t<Stream::is_encoder_stream>>
   Stream &operator<<(Stream &s, const BlockHeader &bh) {
-    return s << bh.parent_hash << bh.number << bh.state_root
-             << bh.extrinsics_root << bh.digests;
+    return s << bh.parent_hash << boost::multiprecision::cpp_int(bh.number)
```

Comment:
usage of aliased `using CompactInteger = boost::multiprecision::cpp_int;` is preferable.